test/server/cards/FateReflection.spec.js

```diff
@@ -0,0 +1,60 @@
+describe('Fate Reflection reaction spell', function () {
+    beforeEach(function () {
+        this.setupTest({
+            player1: {
+                phoenixborn: 'aradel-summergaard',
+                inPlay: ['flute-mage', 'hammer-knight'],
+                spellboard: [],
+                hand: ['anchornaut'],
+                dicepool: ['natural', 'natural', 'charm', 'charm']
+            },
+            player2: {
+                phoenixborn: 'coal-roarkwin',
+                inPlay: ['iron-worker', 'mist-spirit'],
+                spellboard: ['summon-butterfly-monk'],
+                dicepool: ['natural', 'natural', 'charm', 'charm', 'time'],
+                archives: ['butterfly-monk'],
+                hand: ['fate-reflection']
+            }
+        });
+    });
+
+    it('can be played when a unit takes damage', function () {
+        expect(this.ironWorker.tokens.damage).toBeUndefined;
```

Comment:
toBeUndefined is a function that needs to be called. i think the damage should be 0 anyway

Reply:
actually i got confused between `ironWorker.tokens.damage` and `ironWorker.damage`. the token one is undefined now (though you still need to call the function). though that seems odd to me and just using the one that's 0 seems like a better idea IMO

Reply:
if you're reading it, you can do expect(this.ironWorker.damage).toBe(x)
if you're setting it you need ironworker.tokens.damage = x

Many cards don't follow that rule because it was in the early days!

Reply:
Ah, I must have picked an old template for the test too. I'll have a look at better ways of doing this later on.